### PR TITLE
[support spark2.3] newly copyed OAP's unit test codes to support spark2.3

### PR DESCRIPTION
**Pull request by zhixingheyi-tian**
The following unit test code files have been slightly modified based on the same name file in the spark’'s Suite. So with the change of the spark version, there must be a corresponding changes：

src/test/spark2.3/scala/org/apache/spark/sql/execution/datasources/FileSourceStrategySuite.scala
src/test/spark2.3/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetSchemaSuite.scala
src/test/spark2.3/scala/org/apache/spark/sql/execution/vectorized/ColumnarBatchSuite.scala

ParquetDataFileSuite.scala : This code file distinguishes between multiple versions because the methods and constructs it depends on have changed a lot, and the "Apapter" method does not completely solve the problem.

src/test/spark2.3/scala/org/apache/spark/sql/execution/datasources/oap/io/ParquetDataFileSuite.scala

### CODE REVIEW

**Comment by JkSelf** [comment]
Please add the description.

**Comment by carsonwang** [comment]
>  This code file distinguishes between multiple versions because the methods and constructs it depends on have changed a lot, and the "Apapter" method does not completely solve the problem.

@zhixingheyi-tian , can you please explain more what is changed..

**Comment by carsonwang** [comment]
Does this mean we need add these unit tests in spark 2.1 and 2.2 dir also?

File: src/test/spark2.3/scala/org/apache/spark/sql/execution/datasources/oap/io/ParquetDataFileSuite.scala [hunk 1]
```diff
@@ -0,0 +1,652 @@
...
+  }
+
+  test("test load multi-columns every time") {
+    val exception = intercept[IllegalArgumentException] {
+      loadSingleColumn(Array(0, 1))
+    }
+    assert(exception.getMessage.contains("Only can get single column every time"))
+  }
+}
+
```
Comment (zhixingheyi-tian) [inline src/test/spark2.3/scala/org/apache/spark/sql/execution/datasources/oap/io/ParquetDataFileSuite.scala]: 20a21,22
> import java.util
> import java.util.TimeZone
35c37
< import org.apache.parquet.schema.{MessageType, PrimitiveType}
---
> import org.apache.parquet.schema.{MessageType, PrimitiveType, Type}
45d46
< import org.apache.spark.sql.execution.vectorized.{ColumnarBatch, ColumnVector}
48a50
> import org.apache.spark.sql.oap.adapter.ColumnVectorAdapter
50a53
> import org.apache.spark.sql.vectorized.ColumnarBatch
348c351
<       val rowIterator = batch.rowIterator()
---
>       val rowIterator = batch.rowOapIterator()
537c540
<     val vector = ColumnVector.allocate(rowCount, IntegerType, MemoryMode.ON_HEAP)
---
>     val vector = ColumnVectorAdapter.allocate(rowCount, IntegerType, MemoryMode.ON_HEAP)
539a543
>     val types: util.List[Type] = parquetSchema.asGroupType.getFields
543c547,548
<         new VectorizedColumnReader(columnDescriptor, fiberData.getPageReader(columnDescriptor)))
---
>         new VectorizedColumnReader(columnDescriptor, types.get(0).getOriginalType,
>           fiberData.getPageReader(columnDescriptor), TimeZone.getDefault))
556c561
<     val vector = ColumnVector.allocate(rowCount, IntegerType, MemoryMode.ON_HEAP)
---
>     val vector = ColumnVectorAdapter.allocate(rowCount, IntegerType, MemoryMode.ON_HEAP)

**Comment by carsonwang** [comment]
You said they should be a rename for Spark 2.1, why the new files are added?

**Comment by zhixingheyi-tian** [comment]
@carsonwang 
This is because the delete operation was not submitted.

#850  shows the rename operation

**Comment by carsonwang** [comment]
@zhixingheyi-tian , please update this PR as we are going to merge this one instead of that one.

**Comment by carsonwang** [comment]
Merge this first.

**Pull request merged**